google-cloud-translate/src/main/java/com/google/cloud/translate/TranslateOptions.java

```diff
@@ -135,7 +136,19 @@ public TranslateOptions build() {
 
   private TranslateOptions(Builder builder) {
     super(TranslateFactory.class, TranslateRpcFactory.class, builder, new TranslateDefaults());
-    this.apiKey = builder.apiKey != null ? builder.apiKey : getDefaultApiKey();
+    if (builder.apiKey != null) {
+      this.apiKey = builder.apiKey;
+    } else if (System.getenv(ServiceOptions.CREDENTIAL_ENV_NAME) == null) {
+      this.apiKey = getDefaultApiKey();
+    } else {
```

Comment:
I believe this is the case currently working successfully in the integration tests. If I understand correctly, it avoids the backend conflict between the API key and the service account credentials seen in #1405.

I plan to add some additional tests to be sure. Do you know if there already exists a real API key we could use for integration testing?